### PR TITLE
STN-342 : New Traditional Color Pairing

### PR DESCRIPTION
**Pull request by heatherdesigns**
# [STN-342](https://sparkbox.atlassian.net/browse/STN-342) READY FOR REVIEW

## Summary
The work in this PR includes:
1. In the Colorful color pairings, update `Sand` to `Cardinal`.
2. In the Traditional color pairings, add `Blue Jay`.

Reference: [Color Pairing Spreadsheet](https://docs.google.com/spreadsheets/d/1m2Dt4trd2qLZJhsuXidLBtTRovlm4tLwRyG3ic07ANM/edit#gid=0)

_Note:_ The Traditional Cardinal color pairings have been updated to match the values in the [color pairing spreadsheet](https://docs.google.com/spreadsheets/d/1m2Dt4trd2qLZJhsuXidLBtTRovlm4tLwRyG3ic07ANM/edit#gid=0). I connected with Merani to confirm that the spreadsheet is the source of truth before making the changes. `Primary Dark` is the only exception. I did not update it as I recall it was updated last sprint. There is a note in the spreadsheet regarding the value in the codebase.

## Need Review By (Date)
5/15/2020

## Urgency
low

## Steps to Test
1. Run `npm run test` and verify all of our Sass tests continue to pass.
2. The differences between the Traditional themes `Cardinal` and `Blue Jay` color pairings are somewhat subtle. The most notable difference is seen in the hero and carousel components. In the Traditional theme, go to the QA Page. When the `Cardinal` color pairing is implemented, the color of the image overlay is charcoal grey in color. Switch to the `Blue Jay` color pairing by going to `Appearance > Settings > Humsci Traditional > Color Pairing > select `Blue Jay`. Save configuration and go back to the QA Page. The color of the image overlay should be a shade blue.
3. I also recommend testing the application of the `ht-pairing-bluejay` class on a block in the layout builder.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

**Comment by bryanbraun** [comment]
@heatherdesigns, one area where the traditional theme is diverging from colorful is that the [background well](https://github.com/SU-HSDO/suhumsci/blob/5a4a1d1855b340beeb90ebd975edd3b45c31e5ca/docroot/themes/humsci/humsci_basic/src/scss/utilities/_general.scss#L61) needs to be a color-pallet color (tertiary highlight), not a global gray.

It's subtle, but you can see it below the hero in [the figma designs](https://www.figma.com/file/bG42fKKioEcnJ0ORKkLL5E/%E2%9A%99%EF%B8%8F-Traditional-Theme---Working?node-id=6043%3A56).

File: docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.icons.scss [hunk 1]
```diff
@@ -39,6 +39,11 @@
         $icon-color: hb-get-pairing-color($color, 'cardinal', $ht-traditional-pairings);
         background-image: svg(hc-get-icons($icon, $icon-color));
       }
+
+      .hc-pairing-bluejay & {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.icons.scss]: ooo I think this should be ht-pairing

Comment (silverli) [reply docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.icons.scss]: also line 38!! that's my bad I think

**Review by silverli — CHANGES_REQUESTED** [review]
still looking over things in the browser, but I wonder if you can add a few tests for the color pairing in our sass tests?

**Comment by heatherdesigns** [comment]
Updated items since last review:
See screenshots in JIRA [STN-342](https://sparkbox.atlassian.net/browse/STN-342)
- [x] Wells
- [x] Vertical linked card
- [x] Primary nav item hover state
- [x] Secondary nav item hover state
- [x] Local footer dark variant Secondary button, font awesome icons, links and table header background color

File: docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.icons.scss [hunk 2]
```diff
@@ -72,12 +77,25 @@
...
 
     &:hover::after,
     &:focus::after {
-      @include hb-icon-background('ext-link', 'tertiary-highlight');
+      @include hb-themes(('colorful', 'airy')) {
+        @include hb-icon-background('ext-link', 'tertiary-highlight');
+      }
+
+      @include hb-traditional {
+        @include hb-icon-background('ext-link', 'tertiary-highlight-darken-10'); // TODO: wip - stumped!
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.icons.scss]: This works now. I need to remove comments.

Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.icons.scss]: Done!

File: docroot/themes/humsci/humsci_basic/src/scss/components/_main-nav.scss [hunk 3]
```diff
@@ -186,6 +193,8 @@
 
     // Targeting only the first level of navigation on desktop
     .hb-main-nav__menu-lv1 > li > & {
+      @include hb-global-color('color', 'black');
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_main-nav.scss:196]: does this need to be gray-dark for traditional?

File: docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.icons.scss [hunk 4]
```diff
@@ -35,12 +35,12 @@
       $icon-color: hb-get-pairing-color($color, $hb-traditional-default, $ht-traditional-pairings);
       background-image: svg(hc-get-icons($icon, $icon-color));
 
-      .hc-pairing-cardinal & {
+      .ht-pairing-cardinal & {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.icons.scss:38]: 👍

File: docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.icons.scss [hunk 5]
```diff
@@ -35,12 +35,12 @@
...
         $icon-color: hb-get-pairing-color($color, 'cardinal', $ht-traditional-pairings);
         background-image: svg(hc-get-icons($icon, $icon-color));
       }
 
-      .hc-pairing-bluejay & {
+      .ht-pairing-bluejay & {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.icons.scss:43]: 👍

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_font-awesome.scss [hunk 6]
```diff
@@ -18,8 +18,18 @@
   transition: 125ms color ease-out;
   @include hb-pairing-color('color', 'secondary');
 
+  &:hover,
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/utilities/_font-awesome.scss]: hmm will this apply to all elements that font awesome is on, not just links?

File: docroot/themes/humsci/humsci_basic/tests/sass-specs/tools/_functions.color-pairings.scss [hunk 7]
```diff
@@ -72,15 +73,30 @@
...
         $test: hb-get-pairing-color('primary', 'cardinal', $ht-traditional-pairings);
         $expect: #600e0e;
 
-        @include assert-equal($test, $expect, 'The Default color pairing primary color value either does not match or does not exist.');
+        @include assert-equal($test, $expect, 'The Cardinal color pairing primary color value either does not match or does not exist.');
+      }
+    }
+  }
+
+  // BLUE JAY
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/tests/sass-specs/tools/_functions.color-pairings.scss:89]: 🎉 👏

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_font-awesome.scss (hunk 6 above)
Comment (silverli) [reply docroot/themes/humsci/humsci_basic/src/scss/utilities/_font-awesome.scss]: I think it is

File: docroot/themes/humsci/humsci_basic/src/scss/components/_main-nav.scss [hunk 8]
```diff
@@ -173,12 +180,21 @@
     }
 
     &:hover:not(.is-active) {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_main-nav.scss:182]: I think you need this on a focus state too -- when I click the link I see a grey background for a moment. Same thing for secondary nav

**Review by silverli — CHANGES_REQUESTED** [review]
just a couple little things! this is looking great though, and thanks so much for adding a new sass-true test!

File: docroot/themes/humsci/humsci_basic/src/scss/components/_main-nav.scss (hunk 3 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/components/_main-nav.scss:196]: Nope. :smile: I checked the design comps and the very top level of the nav remains black.

top level

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_font-awesome.scss (hunk 6 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/utilities/_font-awesome.scss]: Oh man! You're right. I need to revisit this then. Good catch!

**Comment by heatherdesigns** [comment]
WYSIWYG code snippet for font awesome icons. This covers two different ways in which font awesome icons may be implemented. The first one is created by making a link first, then with the link selected click on the FA styles to add an icon. You will need to remove the text after adding the icon. The second one is found on https://economics-stage.stanford.edu. I could not figure out how to replicate how to create it, but wanted to make sure the styles continued to be available.

```
<p><a href="https://facebook.com"><span class="fontawesome-icon-inline"><span class="fa-3x fa-facebook fab"></span></span></a></p>

<p><span class="fontawesome-icon-inline"><a href="https://facebook.com"><span class="fa-3x fa-facebook fab"></span></a></span></p>
```

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_font-awesome.scss [hunk 9]
```diff
@@ -33,7 +33,12 @@ a {
   > .fab,
   > .fal,
   > .far,
-  > .fas {
+  > .fas,
+  > .fontawesome-icon-inline > .fa,
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/utilities/_font-awesome.scss:37]: nice work!

**Review by silverli — APPROVED** [review]
Looks great!!! Thanks for all your work!

**Pull request merged**